crates/ark/src/lsp/completions/sources/unique/custom.rs

```diff
@@ -248,11 +256,229 @@ mod tests {
             let document = Document::new("library(uti)", None);
             let context = DocumentContext::new(&document, point, None);
 
-            let compls = completions_from_custom_source_impl(&context)
-                .unwrap()
-                .unwrap();
+            let compls = completions_from_custom_source(&context).unwrap().unwrap();
 
             assert!(compls.iter().any(|c| c.label == "utils"));
         })
     }
+
+    #[test]
+    fn test_completion_custom_sys_getenv() {
+        r_test(|| {
+            let name = "ARK_TEST_ENVVAR";
+
+            harp::parse_eval_global(format!("Sys.setenv({name} = '1')").as_str()).unwrap();
```

Comment:
Here and below: Only use global when needed (to create a side effect there or because of a dependency on global var).

```suggestion
            harp::parse_eval_base(format!("Sys.setenv({name} = '1')").as_str()).unwrap();
```